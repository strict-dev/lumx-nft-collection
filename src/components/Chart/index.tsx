import ChartJS from 'chart.js/auto'
import { Bar, Scatter } from 'react-chartjs-2'

ChartJS.defaults.plugins.legend.display = false

const options = {
  scales: {
    y: {
      beginAtZero: true
    },
    x: {
      display: false
    }
  }
}

type ChartProps = {
  type: 'scatter' | 'bar'
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  dataset: any
}

const Chart: React.FC<ChartProps> = ({ dataset, type }) => {
  return (
    <>
      {type === 'scatter' && <Scatter options={options} data={dataset} />}
      {type === 'bar' && (
        <div className="flex items-center gap-2">
          <p className="text-xs font-bold text-neutral-700">VOL</p>
          <div className="w-full h-10">
            <Bar
              options={{
                ...options,
                maintainAspectRatio: false,
                scales: {
                  y: { display: false },
                  x: { display: false }
                }
              }}
              data={dataset}
            />
          </div>
        </div>
      )}
    </>
  )
}

export default Chart
