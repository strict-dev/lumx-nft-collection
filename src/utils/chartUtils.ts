import { ScriptableContext } from 'chart.js'

export const chartOptions = {
  scales: {
    x: {
      display: false
    },
    y: {
      display: false
    }
  },
  plugins: {
    legend: {
      display: false
    }
  }
}

export const datasetConfig = {
  fill: true,
  borderColor: '#4dbc19',
  borderWidth: 2,
  backgroundColor: (ctx: ScriptableContext<'line'>) =>
    createChartGradient(ctx, 'rgba(77, 188, 25, 0.2)', 'rgba(77, 188, 25, 0)'),
  tension: 0.4,
  pointRadius: 0
}

export const createChartGradient = (
  context: ScriptableContext<'line'>,
  startColor: string,
  endColor: string
) => {
  const ctx = context.chart.ctx
  const gradient = ctx.createLinearGradient(0, 0, 0, 200)
  gradient.addColorStop(0, startColor)
  gradient.addColorStop(0.4, endColor)
  return gradient
}
