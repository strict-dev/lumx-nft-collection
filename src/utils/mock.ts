export const data = {
  collectionName: 'Bored Ape Yacht Club',
  token: '0x7b123f53421b1bF8533339BFBdc7C98aA94163db',
  lifetime: '100d',
  items: '10k',
  royalties: '7.5%',
  floor: '145',
  vol24h: {
    total: '1.2M',
    performance: '12.02%'
  },
  vol7d: {
    total: '12.9K',
    performance: '1.29%'
  },
  sales: 2300,
  hypeTrend: 'Very Hyped'
}
