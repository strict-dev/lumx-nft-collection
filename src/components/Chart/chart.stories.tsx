import { ComponentStory, Meta } from '@storybook/react'
import { barsDataset, scatterDataset1 } from 'utils/mock'

import Chart from '.'
export default {
  title: 'Chart',
  component: Chart
} as Meta

const Template: ComponentStory<typeof Chart> = (args) => <Chart {...args} />

export const Scatter = Template.bind({})

Scatter.args = {
  type: 'scatter',
  dataset: scatterDataset1()
}

export const Bar = Template.bind({})

Bar.args = {
  type: 'bar',
  dataset: barsDataset()
}
