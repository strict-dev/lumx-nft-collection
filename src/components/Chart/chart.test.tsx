import { render } from '@testing-library/react'
import { scatterDataset1 } from 'utils/mock'

import Chart from '.'

describe('<Chart />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Chart dataset={scatterDataset1()} type="scatter" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
