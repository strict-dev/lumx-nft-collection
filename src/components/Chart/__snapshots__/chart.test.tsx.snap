// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> should render the heading 1`] = `
<canvas
  height="150"
  role="img"
  width="300"
/>
`;
